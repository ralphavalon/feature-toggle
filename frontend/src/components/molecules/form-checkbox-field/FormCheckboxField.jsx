import React, { useState, useEffect } from 'react';
import { Form } from 'react-bootstrap';
import PropTypes from 'prop-types';

const FormCheckboxField = ({
  label, name, className,
  disabled, value
}) => {

  const [fieldValue, setFieldValue] = useState(value);
  useEffect(() => setFieldValue(value), [value]);

  return (
    <Form.Group className={className}>
      <Form.Check type="checkbox" name={name} label={label} disabled={disabled} checked={!!fieldValue} value={fieldValue} onChange={e => setFieldValue(!fieldValue)} />
    </Form.Group>
  );
};

FormCheckboxField.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
};

export default FormCheckboxField;