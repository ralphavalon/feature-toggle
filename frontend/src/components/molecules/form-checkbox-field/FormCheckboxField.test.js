import React from 'react';
import { shallow } from 'enzyme';

import FormCheckboxField from './FormCheckboxField';

afterEach(() => {
  jest.clearAllMocks();
});

const formCheckboxFieldRequiredProps = {
  label: 'Label',
  name: 'field-name'
};

let formCheckboxFieldOptionalProps = {
  className: 'className',
  disabled: true,
  value: true,
};

describe('FormCheckboxField', () => {
  it('renders correctly with required props', () => {
    const wrapper = shallow(<FormCheckboxField {...formCheckboxFieldRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('renders correctly with required props and optional props', () => {
    const wrapper = shallow(<FormCheckboxField {...{...formCheckboxFieldRequiredProps, ...formCheckboxFieldOptionalProps}} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should change state when onChange', () => {
    const wrapper = shallow(<FormCheckboxField {...formCheckboxFieldRequiredProps} />);
    wrapper.find('FormCheck').simulate('change', { target: { value: true }});
    expect(wrapper.find('FormCheck').prop('value')).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
})

