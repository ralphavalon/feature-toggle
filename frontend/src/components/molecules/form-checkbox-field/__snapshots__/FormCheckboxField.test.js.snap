// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCheckboxField renders correctly with required props 1`] = `
<FormGroup>
  <FormCheck
    checked={false}
    label="Label"
    name="field-name"
    onChange={[Function]}
    type="checkbox"
  />
</FormGroup>
`;

exports[`FormCheckboxField renders correctly with required props and optional props 1`] = `
<FormGroup
  className="className"
>
  <FormCheck
    checked={true}
    disabled={true}
    label="Label"
    name="field-name"
    onChange={[Function]}
    type="checkbox"
    value={true}
  />
</FormGroup>
`;

exports[`FormCheckboxField should change state when onChange 1`] = `
<FormGroup>
  <FormCheck
    checked={true}
    label="Label"
    name="field-name"
    onChange={[Function]}
    type="checkbox"
    value={true}
  />
</FormGroup>
`;
